packages/flutter/test/widgets/navigator_test.dart

```diff
@@ -627,6 +628,39 @@ void main() {
     expect(observations[2].previous, '/A');
   });
 
+  testWidgetsWithLeakTracking('$Route  dispatches memory events', (WidgetTester tester) async {
+    Future<void> createAndDisposeRoute() async {
+      final GlobalKey<NavigatorState> nav = GlobalKey<NavigatorState>();
+      await tester.pumpWidget(
+      MaterialApp(
+        navigatorKey: nav,
+        home: const Scaffold(
+          body: Text('home'),
+        )
+      )
+    );
+      const TestPage page = TestPage(name: 'page');
+      final Route<dynamic> route = page.createRoute(nav.currentContext!);
+      // ignore: invalid_use_of_protected_member
+      route.dispose();
+    }
+
+    final List<ObjectEvent> events = <ObjectEvent>[];
+    void listener(ObjectEvent event) {
+      if (event.object.runtimeType == MaterialPageRoute<void>) {
+        events.add(event);
+      }
+    }
+    MemoryAllocations.instance.addListener(listener);
+
+    await createAndDisposeRoute();
+    expect(events, hasLength(2));
```

Comment:
This is interesting, I would expect 3 events. One for MaterialPageRoute created from home page of MaterialApp. Two for creating and disposing the TestPage

Reply:
ah nvm, the home page will create MaterialPageRoute<dynamic> which will be filter out in the listener

Reply:
yes!